real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/localizers/MultipleIMUtest.java

```diff
@@ -0,0 +1,395 @@
+            MagneticMeasurement mes = ((MagneticMeasurement)m);
+            System.out.println("IMU: " + mes.getIMU());
+            System.out.println("X: " + mes.getmagX());
+            System.out.println("Y: " + mes.getmagY());
+            System.out.println("Z: " + mes.getmagZ());
+            System.out.println("\n====================================================");
+            */
+        }
+    });
+
+    private Subscriber incSub = new Subscriber("test", NodeChannel.IMU_INCLINATION.getMsgPath(), (topicName, m) -> {
+        if(m instanceof IMUInclinationMessage){
+            processMessage(m);
+            System.out.println("Received Inclination Message");
+            /*
+            System.out.println("Inclination Message; Topic is : " + topicName);
+            IMUInclinationMessage mes = ((IMUInclinationMessage)m);
+            System.out.println("IMU: " + mes.getIMU());
+            System.out.println("X: " + mes.getXInclination());
+            System.out.println("Y: " + mes.getYInclination());
+            System.out.println("Z: " + mes.getZInclination());
+            System.out.println("\n====================================================");
+            */
+        }
+    });
+
+    private Subscriber compSub = new Subscriber("test", NodeChannel.IMU_COMPASS.getMsgPath(), (topicName, m) -> {
+        if(m instanceof IMUCompassMessage){
+            processMessage(m);
+            System.out.println("Received Compass Message");
+            /*
+            System.out.println("Compass Message; Topic is : " + topicName);
+            IMUCompassMessage mes = ((IMUCompassMessage)m);
+            System.out.println("IMU: " + mes.getIMU());
+            System.out.println("Heading: " + mes.getCompassHeading());
+            System.out.println("\n====================================================");
+            */
+        }
+    });
+
+    private Subscriber angPosSub = new Subscriber("test", NodeChannel.IMU_ANG_POS.getMsgPath(), (topicName, m) -> {
+        if(m instanceof IMUAngularPositionMessage){
+            processMessage(m);
+            System.out.println("Received Angular Position Message");
+            /*
+            IMUAngularPositionMessage mes = ((IMUAngularPositionMessage)m);
+            System.out.println("IMU: " + mes.getIMU());
+            double[][] rot = mes.getRot();
+            System.out.println("Values");
+            for(int i = 0; i < rot.length; i++) {
+                for(int j = 0; j < rot[i].length; j++) {
+                    System.out.println("rot["+i+"]["+j+"] = " + rot[i][j]);
+                }
+            }
+            System.out.println("\n====================================================");
+            */
+
+        }
+    });
+
+    private static ArrayList<Message> messages = new ArrayList<>();
+
+    double EPS = Math.pow(10, -5);
+    long waitTime = 3000;
+
+    @Test
+    public void testStandardIMUData() {
```

Comment:
Instead of having all the tests in one single method, split them up into different methods with the `@Test` annotation